content/api/servo-board/GPIO.md

```diff
@@ -0,0 +1,68 @@
+---
+title: GPIO
+---
+
+You can use the GPIO pins to read and write individual pins of the servo assembly. These can be used for anything, from microswitches to LEDs. GPIO is only available on pins 2 to 12, because pins 0 and 1 are reserved for communication with our kit.
+
+## Pin Mode
+GPIO pins support 4 different modes, allowing it to work in 4 different ways. A pin can only have 1 mode at a time, and certain pins aren't compatible with certain modes. These Pin modes are stored in an [enum](https://docs.python.org/3/library/enum.html) which needs to be imported to use.
+
+```python
+from robot import PinMode
+``` 
+
+{{% notice note %}}
+The input modes closely resemble those of an Arduino. More information on them can be found in [their docs](https://docs.python.org/3/library/enum.html)
+{{% /notice %}}
+
+
+### `PinMode.INPUT`
+Allows reading the value from the pin
+
+### `PinMode.INPUT_PULLUP`
+Allows reading the value from the pin using the built-in pullup resistor.
```

Comment:
Feels like we need to explain pullup resistors somewhere, or at least link to a good explanation

Reply:
Agreed, although i'm not entirely sure what one is myself, or at least why it's preferable in some cases. Anyone know a good resource / an provide a short explanation? cc @kierdavis

Reply:
https://learn.sparkfun.com/tutorials/pull-up-resistors looks like a good resource, and it also talks about internal pullup resistors inside microcontrollers (like we use here).

A disadvantage is that it contains code example(s) for Arduino, which may be confusing since it's a different syntax to our kit. I don't think it's too much of a problem though - especially if we ensure there's an example using our syntax near to this link.

It's not an easy topic to explain succinctly - I feel like any description I could write would still be a few paragraphs long.